nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -587,27 +601,166 @@ in {
 
     services.nginx = lib.mkIf cfg.configureNginx {
       enable = true;
-      recommendedProxySettings = true; # required for redirections to work
+
+      commonHttpConfig = ''
+        proxy_cache_path /var/cache/nginx levels=1:2 keys_zone=all_cache:10m inactive=7d max_size=1g;
+      '';
+
       virtualHosts."${cfg.localDomain}" = {
         root = "${cfg.package}/public/";
         forceSSL = true; # mastodon only supports https
         enableACME = true;
 
-        locations."/system/".alias = "/var/lib/mastodon/public-system/";
-
         locations."/" = {
           tryFiles = "$uri @proxy";
+          priority = 1100;
```

Comment:
why are you manually setting the priority for each location? the config in the upstream repo doesn't specify priorities.

Reply:
This option is only used on NixOS. Without this in nginx configuration file `locations` are scattered randomly.